modules/integrations/arrow-integrations-retrofit-adapter/src/main/kotlin/arrow.integrations.retrofit.adapter/Async2CallAdapterFactory.kt

```diff
@@ -0,0 +1,38 @@
+package arrow.integrations.retrofit.adapter
+
+import arrow.effects.IO
+import retrofit2.CallAdapter
+import retrofit2.Retrofit
+import java.lang.reflect.ParameterizedType
+import java.lang.reflect.Type
+
+class Async2CallAdapterFactory : CallAdapter.Factory() {
+
+  companion object {
+    fun create() = Async2CallAdapterFactory()
```

Comment:
All return types in Arrow are always fully typed and we don't rely on inference.